src/main/java/com/checkmarx/flow/service/GitHubService.java

```diff
@@ -82,50 +84,66 @@ void processPull(ScanRequest request, ScanResults results) throws GitHubClientEx
         JsonNode root = objMapper.readTree(response.getBody());
         Iterator<JsonNode> it = root.getElements();
         while (it.hasNext()) {
             JsonNode commentNode = it.next();
             RepoComment comment = createRepoComment(commentNode);
-            if (isCheckMarxComment(comment)) {
+            if (PullRequestCommentsHelper.isCheckMarxComment(comment)) {
                 result.add(comment);
             }
         }
         return result;
     }
 
-    private boolean isCheckMarxComment(RepoComment comment) {
-        return comment.getComment().contains("Full Scan Details") && comment.getComment().contains("Checkmarx scan completed") ||
-                comment.getComment().contains("Scan submitted to Checkmarx");
+    public void deleteComment(String url) {
+        HttpEntity<?> httpEntity = new HttpEntity<>(createAuthHeaders());
+        restTemplate.exchange(url, HttpMethod.DELETE, httpEntity, String.class);
     }
 
-    private RepoComment createRepoComment(JsonNode commentNode) {
+
+    private RepoComment createRepoComment(JsonNode commentNode)  {
         String commentBody = commentNode.path("body").getTextValue();
         long id = commentNode.path("id").asLong();
-        return new RepoComment(id, commentBody);
+        String commentUrl = commentNode.path(("url")).asText();
+        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'");
+        sdf.setTimeZone(TimeZone.getTimeZone("UTC"));
+        String updatedStr = commentNode.path("updated_at").asText();
+        String createdStr = commentNode.path("created_at").asText();
+        try {
+            return new RepoComment(id, commentBody, commentUrl, sdf.parse(createdStr), sdf.parse(updatedStr));
+        }
+        catch (ParseException pe) {
+            throw new GitHubClientRunTimeException("Error parsing github pull request created or updted date", pe);
+        }
     }
 
     public void sendMergeComment(ScanRequest request, String comment) throws GitHubClientException {
         try {
-            List<RepoComment> repoComments = getComments(request.getMergeNoteUri());
-            log.debug("There are {} checkmarx comments on this pull request", repoComments.size());
-            addComment(request, comment);
+            RepoComment commentToUpdate = PullRequestCommentsHelper.getCommentToUpdate(getComments(request.getMergeNoteUri()), comment);
+            if (commentToUpdate !=  null) {
+                if (!PullRequestCommentsHelper.shouldUpdateComment(comment, commentToUpdate.getComment())) {
+                    log.debug("Comment should not be updated");
+                    return;
+                }
+                log.debug("Going to update GitHub pull request comment");
+                updateComment(commentToUpdate.getCommentUrl(), comment);
+            } else {
+                log.debug("Going to create a new GitHub pull request comment");
+                addComment(request, comment);
+            }
```

Comment:
need to make sure exception in write comment attempt would is caught and not terminate the entire pull request update process